src/ToastHeader.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { mapToCssModules, tagPropType } from './utils';
+
+const propTypes = {
+  tag: tagPropType,
+  icon: PropTypes.string,
+  wrapTag: tagPropType,
+  toggle: PropTypes.func,
+  className: PropTypes.string,
+  cssModule: PropTypes.object,
+  children: PropTypes.node,
+  closeAriaLabel: PropTypes.string,
+  charCode: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  close: PropTypes.object,
+};
+
+const defaultProps = {
+  tag: 'strong',
+  wrapTag: 'div',
+  tagClassName: 'mr-auto',
+  closeAriaLabel: 'Close',
+  charCode: 215,
+};
+
+const ToastHeader = (props) => {
+  let closeButton;
+  const {
+    className,
+    cssModule,
+    children,
+    toggle,
+    tag: Tag,
+    wrapTag: WrapTag,
+    closeAriaLabel,
+    charCode,
+    close,
+    tagClassName,
+    icon,
+    ...attributes } = props;
+
+  const classes = mapToCssModules(classNames(
+    className,
+    'toast-header'
+  ), cssModule);
+
+  if (!close && toggle) {
+    const closeIcon = typeof charCode === 'number' ? String.fromCharCode(charCode) : charCode;
+    closeButton = (
+      <button type="button" onClick={toggle} className={mapToCssModules('close', cssModule)} aria-label={closeAriaLabel}>
+        <span aria-hidden="true">{closeIcon}</span>
+      </button>
+    );
+  }
+
+  return (
+    <WrapTag {...attributes} className={classes}>
+      {icon && (
+        <svg
```

Comment:
I think this svg was just for the examples in the bootstrap docs. It would be good to allow the user to specific something custom.
Though, I do like the idea of specifying a color to have a square of that color. Maybe allow a component or a string. In the case of a component, the component is used as is. When it is a string, this svg can be used.